src/__tests__/unwrap.test.tsx

```diff
@@ -0,0 +1,63 @@
+import { z } from "zod";
+import React from "react";
+import { createTsForm } from "../createSchemaForm";
+
+describe("unwrap types", () => {
+  it("should have the correct types for nullable, optional, and nullish", () => {
+    function TextField(_: { one: string }) {
+      return <div />;
+    }
+    const mapping = [[z.string(), TextField]] as const;
+
+    const Form = createTsForm(mapping);
+
+    <Form
+      onSubmit={(data) => {
+        data.nullableString;
```

Comment:
Do you also want to add

`data.optionalString` and
`data.nullishString` here?